packages/flutter_tools/lib/src/flx.dart

```diff
@@ -115,7 +115,13 @@ Map<_Asset, List<_Asset>> _parseAssets(
 
       List<FileSystemEntity> files = assetDir.listSync(recursive: true);
 
+      final String buildDirPath = path.absolute(path.join(assetBase, 'build'));
```

Comment:
> I just want to ensure that the FLX working directory will be excluded if its path was overridden by a command line flag

Ah, ok.

> Alternatively, maybe we should require that assets be located within a dedicated directory? Or require that variants follow a specific file layout or be declared explicitly in the manifest?
> Doing a search through the entire directory tree for files matching a given name seems like it carries the risk of misidentifying variants.

Totally agree. I'll change this PR to exclude both the `build/` dir, and the directory that we're zipping up (which, they'll frequently be overlap there).

Also cc'ing @krisgiesing fr ideas re: adopting some conventions around where assets and their variants can live.